dev/camdenhells/REQUIREMENTS.md

```diff
@@ -44,7 +44,7 @@ For reviews and updates:
     * Does not imply support for the entirety of that version's API
     * Does imply that engine features available in earlier versions are still available
   * Using an engine feature that depends on a version greater than is available on a device:
-    * Will in general throw a runtime error exception
+    * Will in general throw a checked exception, forcing the user code to deal with it. 
```

Comment:
I would prefer to avoid checked exceptions except when it is an absolutely necessary part of the API contract, i.e. for something that cannot be otherwise reliably tested/predicted. I believe this approach reduces client code clutter, and discourages both client use of exceptions for normal control flow and the client just re-throwing all Exceptions to avoid the try/catch code when there are numerous checked exceptions that rarely occur, or that never occur if the contract is observed.

In this case, that would suggest that we should either require that the client checks versions before using a version-specific feature (as called out in the next bullet), or throw a checked exception, but not both, and since it is predictable, I would say the former. 

When we resolve this, we should probably add a section to the coding conventions, as this is a general issue.

Reply:
Perhaps I should have been clearer. I wish to throw checked exceptions for things that are not catostophic. For example, texture operations. Nothing about a texture operation failing should prevent attempting to fix the problem. For this reason, the programmer should be forced at compile time to make a decision to either gracefully degrade/recover or if they are being lazy, bury the exception and deal with whatever the render behavior becomes.

For this where no recovery is possible, I don't wish to throw checked exceptions.